trimesh/path/path.py

```diff
@@ -918,34 +921,54 @@ def to_planar(self,
         if to_2D is None:
             # fit a plane to our vertices
             C, N = plane_fit(self.vertices[referenced])
-            # apply the normal sign hint
+            # apply a passed normal
             if normal is not None:
                 normal = np.asanyarray(normal, dtype=np.float64)
                 if normal.shape == (3,):
-                    N *= np.sign(np.dot(N, normal))
-                    N = normal
+                    dn = normal
                 else:
                     log.warning(
                         "passed normal not used: {}".format(
                             normal.shape))
-            # create a transform from fit plane to XY
+            # otherwise use normal from fit plane
+            else:
+                dn = N
+            # eliminate calculation errors
+            C, N, dn = util.snap_to(C), util.snap_to(N), util.snap_to(dn)
+            # create a transform from fit plane to destination plane
             to_2D = plane_transform(origin=C,
-                                    normal=N)
+                                    normal=N,
+                                    destination_normal=dn)
+        elif normal is not None:
+            log.warning("passed normal not used: {}".format(
+                normal.shape))
 
         # make sure we've extracted a transform
         to_2D = np.asanyarray(to_2D, dtype=np.float64)
         if to_2D.shape != (4, 4):
             raise ValueError('unable to create transform!')
 
+        # make sure normal aligns with X, Y, or Z axis
+        normal_test = False
+        for i in [[1, 0, 0], [0, 1, 0], [0, 0, 1],
```

Comment:
Probably an easier way to do this check is with `np.isclose`, which you can pass `atol` and `rtol` values for float comparisons:
```
In [1]: import numpy as np

In [2]: normal = [0,0,1.0]

In [3]: np.isclose(np.abs(normal), np.eye(3))
Out[3]: 
array([[False,  True, False],
       [ True, False, False],
       [ True,  True,  True]])

In [5]: check = np.isclose(np.abs(normal), np.eye(3)).all(axis=1)

In [6]: check
Out[6]: array([False, False,  True])
```